internal/codegen/tpl_stencil_schema.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2025 stencil contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package codegen
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"errors"
+
+	"github.com/santhosh-tekuri/jsonschema/v6"
+)
+
+// validateJSONSchema validates the provided data against the provided
+// schema.
+//
+// If "identifier" is provided its used as a friendly name for the JSON
+// schema in error messages.
+func validateJSONSchema(identifier string, schemaMap map[string]any, data any) error {
```

Comment:
I feel like this needs some unit tests ..

Reply:
It has this today, but not directly testing this, but you're right. I'll get some direct tests of this tomorrow